### PR TITLE
Validate email without any special character

### PR DESCRIPTION
**Pull request by hoan-bitholla**

### CODE REVIEW

File: server/api/controllers/user.js [hunk 1]
```diff
@@ -407,7 +409,7 @@ const requestResetPassword = (req, res) => {
 			let errorMessage = errorMessageConverter(err);
 
 			if (errorMessage === USER_NOT_FOUND) {
-				errorMessage = `Password request sent to: ${email}`;
+				errorMessage = `Email ${email} not found`;
```
Comment (abeikverdi) [inline server/api/controllers/user.js:412]: @hoan-bitholla these messages are done this way so the user sending the request cant tell whether or not an email exists in the system. The returned message should be the same regardless.

File: server/api/controllers/user.js [hunk 2]
```diff
@@ -388,13 +388,15 @@ const requestResetPassword = (req, res) => {
 		domain
 	);
 
-	if (typeof email !== 'string' || !isEmail(email)) {
+	const emailPattern = /^[a-zA-Z0-9_\-.]+@[a-zA-Z0-9\-]+\.[a-zA-Z0-9\-.]+$/;
```
Comment (abeikverdi) [inline server/api/controllers/user.js:391]: The isEmail library is doing the same thing. It doesnt seem necessary to do regex here.

File: server/api/controllers/user.js (hunk 1 above)
Comment (hoan-bitholla) [reply server/api/controllers/user.js:412]: OK, sure! I'll revert it

File: server/api/controllers/user.js (hunk 2 above)
Comment (hoan-bitholla) [reply server/api/controllers/user.js:391]: I checked isEmail function, its allow some special characters, e.g: !, #, %, &, ...

`var emailUserPart = /^[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~]+$/i;
var quotedEmailUser = /^([\s\x01-\x08\x0b\x0c\x0e-\x1f\x7f\x21\x23-\x5b\x5d-\x7e]|(\\[\x01-\x09\x0b\x0c\x0d-\x7f]))*$/i;
var emailUserUtf8Part = /^[a-z\d!#\$%&'\*\+\-\/=\?\^_`{\|}~\u00A0-\uD7FF\uF900-\uFDCF\uFDF0-\uFFEF]+$/i;`

Comment (hoan-bitholla) [reply server/api/controllers/user.js:391]: I checked isEmail function, its allow some special characters, e.g: !, #, %, &, ...

https://github.com/validatorjs/validator.js/blob/8c4b3b35c3bd36dc3983aadb045422847154451b/src/lib/isEmail.js#L19-L25

Comment (hoan-bitholla) [reply server/api/controllers/user.js:391]: I've checked isEmail function, its allow some special characters, e.g: !, #, $, %, &, ...
image

Comment (brandonkimmmm) [reply server/api/controllers/user.js:391]: There are [certain special characters](https://www.w3resource.com/javascript/form/email-validation.php) that should be allowed in emails. For example, we use the plus sign `+` when we want to create additional accounts using the same email ex. `brandon+testing@bitholla.com`. I think at this time, we should use the `isEmail` function from `validator` library. The library still validates emails with rare invalid characters (`£`, `¶`, `§`, etc.) and there  is a [better library for email validation](https://www.npmjs.com/package/email-validator) but we would need to install it on the Kit repo.

**Pull request closed without merging**